tests/e2e/upgradecluster/upgradecluster_test.go

```diff
@@ -0,0 +1,418 @@
+					g.Expect(node.Status).Should(Equal("Ready"))
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParseNodes(kubeConfigFile, true)
+			fmt.Printf("\nFetching Pods status\n")
+			Eventually(func(g Gomega) {
+				pods, err := e2e.ParsePods(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "helm-install") {
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
+					} else {
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
+					}
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+
+		It("Verify ClusterIP Service", func() {
+			_, err := e2e.DeployWorkload("clusterip.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("Cluster IP manifest not deployed", err)
+			}
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-clusterip --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+				res, _ := e2e.RunCommand(cmd)
+				g.Expect(res).Should(ContainSubstring("test-clusterip"))
+			}, "240s", "5s").Should(Succeed())
+
+			clusterip, _ := e2e.FetchClusterIP(kubeConfigFile, "nginx-clusterip-svc")
+			cmd := "\"curl -L --insecure http://" + clusterip + "/name.html\""
+			fmt.Println(cmd)
+			for _, element := range serverNodeNames {
+				nodeName := strings.TrimSpace(element)
+				if nodeName == "" {
+					continue
+				}
+				res, _ := e2e.RunCmdOnNode(cmd, nodeName)
+				fmt.Println(res)
+				Eventually(res).Should(ContainSubstring("test-clusterip"))
+			}
+		})
+
+		It("Verify NodePort Service", func() {
+			_, err := e2e.DeployWorkload("nodeport.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("NodePort manifest not deployed", err)
+			}
+			for _, element := range serverNodeNames {
+				nodeName := strings.TrimSpace(element)
+				if nodeName == "" {
+					continue
+				}
+				node_external_ip, _ := e2e.FetchNodeExternalIP(nodeName)
+				cmd := "kubectl get service nginx-nodeport-svc --kubeconfig=" + kubeConfigFile + " --output jsonpath=\"{.spec.ports[0].nodePort}\""
+				nodeport, _ := e2e.RunCommand(cmd)
+				cmd = "curl -L --insecure http://" + node_external_ip + ":" + nodeport + "/name.html"
+				fmt.Println(cmd)
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+				Eventually(func(g Gomega) {
+					cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-nodeport --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+					res, _ := e2e.RunCommand(cmd)
+					g.Expect(res).Should(ContainSubstring("test-nodeport"), "nodeport pod was not created")
+				}, "240s", "5s").Should(Succeed())
+
+			}
+		})
+
+		It("Verify LoadBalancer Service", func() {
+			_, err := e2e.DeployWorkload("loadbalancer.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println(err, "loadbalancer manifest not deployed")
+			}
+			for _, element := range serverNodeNames {
```

Comment:
As above, we should need to be stripping or checking for empty names. All these for loops can be cleaned up.